### PR TITLE
PB-426 update examples and docs

### PR DESCRIPTION
**Pull request by janpetschexain**
### References

- [PB-426](https://xainag.atlassian.net/browse/PB-426)

### Summary

- use `init_weights()` for the sdk examples
- update documentation accordingly

### Are there any open tasks/blockers for the ticket?

- [the mr](https://github.com/xainag/xain-sdk/pull/86) with the changes to the sdk must be merged first, until then the examples won't be executable

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [ ] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [ ] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [ ] Linked the ticket in the merge request title or the references section.
- [ ] Added an informative merge request summary.

**Code checklist**

- [ ] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [ ] Passed scope checks.
- [ ] Added or updated tests if needed.
- [ ] Added or updated code documentation if needed.
- [ ] Conforms to Google docstring style.
- [ ] Conforms to XAIN structlog style.

### CODE REVIEW

File: docs/examples/pytorch.md [hunk 1]
```diff
@@ -120,7 +121,7 @@ self.model_shapes: List[Tuple[int, ...]] = self.get_tensorflow_shapes(model=self
 
 ## Pytorch Data
 
-The data on which the model will be trained, can either be loaded from a data source (e.g. file, bucket, data base) during the initialization of the `Participant` or on the fly in a `train_round()`. Here, we employ randomly generated placeholder data as an example. This is by no means a meaningful dataset, but it should be sufficient to convey the overall idea.
+The data on which the model will be trained, can either be loaded from a data source (e.g. file, bucket, data base) during the initialization of the `Participant` or on the fly in a `train_round()`. Here, we employ randomly generated placeholder data as an example, which is wrapped in the `init_datasets()` helper function. This is by no means a meaningful dataset, but it should be sufficient to convey the overall idea.
```
Comment (finiteprods) [inline docs/examples/pytorch.md]:
```suggestion
The data on which the model will be trained, can either be loaded from a data source (e.g. file, bucket, database) during the initialization of the `Participant` or on the fly in a `train_round()`. Here, we employ randomly generated placeholder data as an example, which is wrapped in the `init_datasets()` helper function. This is by no means a meaningful dataset, but it should be sufficient to convey the overall idea.
```

File: docs/examples/pytorch.md [hunk 2]
```diff
@@ -180,17 +181,22 @@ self.testset: DataLoader = DataLoader(
...
+
 The implementation of the actual `train_round()` method consists of three main steps. First, the provided `weights` of the global model are loaded into the local model, as
 
 ```python
-if weights is not None:
-    self.set_pytorch_weights(weights=weights, shapes=self.model_shapes, model=self.model)
+self.set_pytorch_weights(weights=weights, shapes=self.model_shapes, model=self.model)
 ```
 
 Next, the local model is trained for certain `epochs` on the local data, whereby the metrics are gathered in each epoch, as
```
Comment (finiteprods) [inline docs/examples/pytorch.md]:
```suggestion
Next, the local model is trained for a certain number of `epochs` on the local data, whereby the metrics are gathered in each epoch, as
```

Comment (finiteprods) [reply docs/examples/pytorch.md]: similar with the other TensorFlow keras example

File: examples/pytorch/example.py [hunk 3]
```diff
@@ -116,38 +127,30 @@ def train_round(
...
+                self.model.optimizer.zero_grad()
+                self.model.loss(self.model(data), label).backward()
+                self.model.optimizer.step()
+            loss: float = 0.0
+            accuracy: float = 0.0
+            for data, label in self.valset:
+                prediction: Tensor = self.model(data)
+                loss += self.model.loss(prediction, label).item()
+                accuracy += (prediction.argmax(dim=1) == label).sum().float()
+            accuracy /= self.val_samples
```
Comment (finiteprods) [inline examples/pytorch/example.py]: can you briefly explain this change (for the non-ML expert)?

Comment (janpetschexain) [reply examples/pytorch/example.py]: basically, pytorch doesn't have any kind of machine learning metrics in its standard library, which means one either has to import these from other 3rd party libraries or implement it by hand. the latter was the faster ways for the accuracy (for more involved metrics it probably makes sense to look for and import stuff).

the two lines of code count the number of correct predictions and divides this by the total number of samples to calculate the overall accuracy.

File: docs/examples/pytorch.md (hunk 1 above)
Comment (janpetschexain) [reply docs/examples/pytorch.md]: changed.

File: docs/examples/pytorch.md (hunk 2 above)
Comment (janpetschexain) [reply docs/examples/pytorch.md]: changed.
